experimental/cuda2/cuda_device.c

```diff
@@ -531,8 +584,28 @@ static iree_status_t iree_hal_cuda2_device_queue_execute(
     const iree_hal_semaphore_list_t signal_semaphore_list,
     iree_host_size_t command_buffer_count,
     iree_hal_command_buffer_t* const* command_buffers) {
-  return iree_make_status(IREE_STATUS_UNIMPLEMENTED,
-                          "queue execution not yet implmeneted");
+  iree_hal_cuda2_device_t* device = iree_hal_cuda2_device_cast(base_device);
+
+  // TODO(benvanik): trace around the entire submission.
+
+  for (iree_host_size_t i = 0; i < command_buffer_count; i++) {
+    CUgraphExec exec =
+        iree_hal_cuda2_graph_command_buffer_handle(command_buffers[i]);
+    IREE_CUDA_RETURN_IF_ERROR(device->cuda_symbols,
+                              cuGraphLaunch(exec, device->cu_stream),
+                              "cuGraphLaunch");
+  }
+
+  // TODO(antiagainst): implement semaphores - for now this conservatively
+  // synchronizes after every submit.
+  IREE_TRACE_ZONE_BEGIN_NAMED(z0, "cuStreamSynchronize");
+  IREE_CUDA_RETURN_IF_ERROR(device->cuda_symbols,
```

Comment:
has to be status = here so that the trace zone end is hit (otherwise the zone won't be ended and tracing tools will explode)